### PR TITLE
Select: Improve usability slightly

### PR DESCRIPTION
**Pull request by torkelo**
Inspired by design of antd select: https://ant.design/components/select/

* Makes it move obvious that you can type / search for values in the Select by switching icon to search icon, and de-emphasize the current value (makes the blinking input cursor more visible)
* Improve the custom value option label, remove "Create: " prefix and add icons to help user understand that they can just use enter key to specify the value 

The custom value label change might not make that much of a difference, could make it worse. Hard to say 

https://user-images.githubusercontent.com/10999/163462289-c0a7a6f3-eb6e-4671-b9af-84e61e78205b.mp4

### CODE REVIEW

File: packages/grafana-ui/src/components/Select/DropdownIndicator.tsx [hunk 1]
```diff
@@ -6,6 +6,7 @@ interface DropdownIndicatorProps {
 }
 
 export const DropdownIndicator: React.FC<DropdownIndicatorProps> = ({ isOpen }) => {
-  const icon = isOpen ? 'angle-up' : 'angle-down';
-  return <Icon name={icon} />;
+  const icon = isOpen ? 'search' : 'angle-down';
```
Comment (darrenjaneczek) [inline packages/grafana-ui/src/components/Select/DropdownIndicator.tsx:9]: Can we define a `searchingIcon` or `typingIcon` property which is defaulted to "search"?

Comment (torkelo) [reply packages/grafana-ui/src/components/Select/DropdownIndicator.tsx:9]: not sure that is a good idea. Think having different icons in different selects would be confusing

Comment (torkelo) [reply packages/grafana-ui/src/components/Select/DropdownIndicator.tsx:9]: or maybe I misunderstood you, what did you mean?

**Comment by lukasztyrala** [comment]
> - Makes it move obvious that you can type / search for values in the Select by switching icon to search icon, and de-emphasize the current value (makes the blinking input cursor more visible)

This is an improvement to previous interactions as it communicates more clearly that: users can type (cursor more visible), that there is something selected that will be overwritten (greyed label inside input), and that by typing search will happen (icon change to the magnifying glass).

It is important though that the magnifying glass is OK only in a scenario like this. When we have a regular drop-down, where people can just pick one of the options from the list, and are unable to search by typing then the magnifying glass might be confusing (and we should stick with the arrow up/down pattern).

> - Improve the custom value option label, remove "Create: " prefix and add icons to help user understand that they can just use enter key to specify the value

This on the other hand will be hard to decode, as a lot of things happen here simultaneously:
1. The component needs to communicate, that nothing on the list matches entered string.
2. That you can consider transforming entered string (for the purpose of search) into a new item.
3. That to add the new item user needs to click Enter.

Showing a keyboard icon and Enter might not be clear enough, as this is a convention that we use to communicate shortcuts. Communicating a shortcut requires that people know about function existence. In this example, people might not be aware of point 2 above (that they can add new things). If you do not know that you can add something, having a shortcut is not helping, because you do not know what is that shortcut for – it might be still associated with hitting enter to search.

The possible alternative can include more icon changes. With the proposed change we already use icon change to communicate that component behaviour changed (_open drop-down_ → _now you can search_ → _now you can add this_):

states-illustration-not-a-final-design

⚠️ This is not a _final_ design, only an illustration of an argument, and requires further details polishing! :-)

**Comment by staton-hyse11** [comment]
Some things I'm thinking:
1. Changing the icon on click feels disruptive and potentially misleading by presenting what looks like a typical dropdown and revealing it behaves differently after a user interacts with it.

2. Icon-only presentations are open to interpretation and have a lot to communicate to the users, as @lukasztyrala mentioned. Providing some text (like Luke's suggestion) will provide clarity and direction for users. 

3. With iterative improvements on this component, I would love to be able to A/B test the differences with live users. Monitoring use and metrics will prove what option works best for our customer base and we can implement the deemed winner to 100% of the Grafana population. This will help root our components in research and tested scenarios instead of taking chances and hurting experiences. I'm not sure what, if any, tools we have for this but will investigate.

**Comment by lukasztyrala** [comment]
> With iterative improvements on this component, I would love to be able to A/B test the differences with live users. […]

As this is an oss change, running an a/b test is very hard to pull, especially since it is a complex interaction pattern (a lot of things happen at once, and many factors impact interactions, other than the component design itself). We can always run usability sessions (even unmoderated), as they provide more nuanced observations and insight into peoples' context. :-)

**Comment by torkelo** [comment]
@lukasztyrala @staton-hyse11 

Thanks for the feedback and luke I like your suggestion for the create the workflow of making the option read "Nothing found. Hit enter to add new value"  and not be styled as an option, that is better than showing the icon keyboard + enter key icon. Will see how easy this is to implement. 

> Changing the icon on click feels disruptive and potentially misleading by presenting what looks like a typical dropdown and revealing it behaves differently after a user interacts with it.

Don't think it's that disruptive, and it's actually kind of accurate. With these kinds of selects that turn into inputs when you open them, they do change character when you interact with them. So changing the icon makes this transition more clear. They are not native selects. 

I think users are most familiar with these types of select & input combo components as all component libraries have some form of them.

**Comment by aocenas** [comment]
Also agree with @lukasztyrala suggestions, the shortcut only does not seem enough if you don't already know how the select work.

Btw does this translate to the cascade component? Would assume so as it basically switches to Select when searching but didn't try it.

**Comment by torkelo** [comment]
Sadly the "No options found. Hit enter to add value" message is not working well as it will be shown even when there are options found. The select library we are using allow me to customize the "Create new option" option, but I cannot make this option appear different then the current search term (Value to create/add) matches options vs when it does not. 

Screenshot from 2022-04-20 08-45-56

**Comment by torkelo** [comment]
I think showing the value you entered along with a "Hit enter to add" hint on the right side could be a good option as well? 

Screenshot from 2022-04-20 08-49-05
Screenshot from 2022-04-20 08-49-55
Screenshot from 2022-04-20 08-50-50

**Review by ivanahuckova — APPROVED** [review]
LGTM!

Just a thought - I was thinking that for Prometheus metric selector, we could also change `placeholder` from `Select metric` to `Search metric` that could also have some effect to make it more obvious that you can search there as well. 🤔

**Review by aocenas — APPROVED** [review]

**Pull request merged**